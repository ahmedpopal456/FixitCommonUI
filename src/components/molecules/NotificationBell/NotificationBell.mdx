---
name: NotificationBell
menu: 2 - Molecules
---
import { Props, Playground } from "docz";
import { NotificationBell } from "./NotificationBell.tsx";
import { DoczContainer } from "../../../utils/DoczContainer.tsx"

# NotificationBell
The NotificationBell displays the notification count as a badge. It performs
a user defined action when pressed. The `notifications` and `onPress` props are
required.

# Usage
```tsx
import { NotificationBell } from "fixit-common-ui";
<NotificationBell notifications={0} onPress={() => {}}/>
```

## Props
<Props of={NotificationBell}></Props>

## Default
<Playground>
<DoczContainer>
  <NotificationBell notifications={2} onPress={() => {}}/>
  <NotificationBell notifications={0} onPress={() => {}}/>
</DoczContainer>
</Playground>
