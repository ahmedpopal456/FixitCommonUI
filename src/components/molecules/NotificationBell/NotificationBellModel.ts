export interface NotificationBellProps {
  /** Callback function to be called when pressed */
  onPress(): void;
  /** Number of notifications to display in the red bubble */
  notifications: number;
  testID?: string;
}
