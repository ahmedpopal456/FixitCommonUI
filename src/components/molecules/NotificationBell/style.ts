import styled from "styled-components/native";
import { Label } from "../../atoms/Typography";
import { Icon } from "../../atoms/Icon";
import colors from "../../../theme/colors";

interface NotificationIconProps {
  notifications: number;
}

export const Container = styled.View`
  width: 80px;
  height: 80px;
`;
export const StyledIcon = styled(Icon)`
  ${(notificationIconProps: NotificationIconProps) =>
    notificationIconProps.notifications > 0 &&
    `
        transform: rotate(-20deg);
  `}
`;

export const Counter = styled(Label)`
  position: absolute;
  top: 10px;
  right: 10px;
  min-width: 25px;
  min-height: 25px;
  font-size: 14px;
  text-align: center;
  color: white;
  font-weight: bold;
  border-radius: 20px;
  background-color: ${colors.notification};
`;
