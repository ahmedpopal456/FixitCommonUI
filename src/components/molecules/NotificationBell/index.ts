import { NotificationBell } from "./NotificationBell";
export { NotificationBell };
