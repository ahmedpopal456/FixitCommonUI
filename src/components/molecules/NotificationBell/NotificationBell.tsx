import React from "react";
import { Button } from "../../atoms/Button";
import { StyledIcon, Counter, Container } from "./style";
import { NotificationBellProps } from "./NotificationBellModel";

/**
 * The NotificationBell displays the notification count as a badge. It performs
 * a user defined action when pressed. The `notifications` and `onPress` props
 * are required.
 *
 * @param props - The NotificationBell's props
 * @returns A react native custom NotificationBell component
 */
export const NotificationBell: React.FC<NotificationBellProps> = (
  notificationbellProps: NotificationBellProps
): JSX.Element => {
  const notificationCount = notificationbellProps.notifications.toString();

  return (
    <Container {...notificationbellProps}>
      {notificationbellProps.notifications > 0 && (
        <Counter>{notificationCount}</Counter>
      )}
      <Button
        onPress={notificationbellProps.onPress}
        color="transparent"
        padding={0}
      >
        <StyledIcon
          notifications={notificationbellProps.notifications}
          library="FontAwesome5"
          name="bell"
          size={45}
        />
      </Button>
    </Container>
  );
};
