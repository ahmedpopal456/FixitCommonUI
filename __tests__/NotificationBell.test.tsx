import "react-native";
import React from "react";
import { NotificationBell } from "../src/components/molecules/NotificationBell";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react-native";

it("renders correctly", () => {
  renderer.create(
    <NotificationBell notifications={0} onPress={() => {}}></NotificationBell>
  );
});

/** Testing user interaction */
it("fires onPress event when pressed", () => {
  const mockFn = jest.fn();

  const { getByTestId } = render(
    <NotificationBell
      notifications={0}
      onPress={mockFn}
      testID="notificationsBellTestId"
    ></NotificationBell>
  );
  const notificationBell = getByTestId("notificationsBellTestId");

  fireEvent.press(notificationBell);

  expect(mockFn).toHaveBeenCalled();
});

it("matches no notifications snapshot", () => {
  const mockFn = jest.fn();
  const button = renderer.create(
    <NotificationBell notifications={0} onPress={mockFn}></NotificationBell>
  );

  expect(button.toJSON()).toMatchSnapshot();
});

it("matches notifications snapshot", () => {
  const mockFn = jest.fn();
  const button = renderer.create(
    <NotificationBell notifications={8} onPress={mockFn}></NotificationBell>
  );

  expect(button.toJSON()).toMatchSnapshot();
});
