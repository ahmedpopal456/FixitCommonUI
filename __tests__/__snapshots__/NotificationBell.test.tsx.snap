// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches no notifications snapshot 1`] = `
<View
  notifications={0}
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "height": 80,
        "width": 80,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#00000000",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "opacity": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        library="Ionicons"
        notifications={0}
        style={
          Array [
            Object {
              "color": "#33343E",
              "fontSize": 45,
            },
            Array [
              Object {},
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`matches notifications snapshot 1`] = `
<View
  notifications={8}
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "height": 80,
        "width": 80,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      color="dark"
      fontSize={13}
      fontWeight={700}
      justification="left"
      padding={3}
      style={
        Array [
          Object {
            "color": "#1D1F2A",
            "flexDirection": "row",
            "fontFamily": "Raleway",
            "fontSize": 13,
            "fontWeight": "700",
            "paddingBottom": 3,
            "paddingLeft": 3,
            "paddingRight": 3,
            "paddingTop": 3,
            "textAlign": "left",
          },
          Object {
            "backgroundColor": "#FF0000",
            "borderRadius": 20,
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "minHeight": 25,
            "minWidth": 25,
            "position": "absolute",
            "right": 10,
            "textAlign": "center",
            "top": 10,
          },
        ]
      }
    >
      8
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#00000000",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "opacity": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        library="Ionicons"
        notifications={8}
        style={
          Array [
            Object {
              "color": "#33343E",
              "fontSize": 45,
            },
            Array [
              Object {
                "transform": Array [
                  Object {
                    "rotate": "-20deg",
                  },
                ],
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
